battle_move_test.go

```diff
@@ -23,3 +23,43 @@ func NewMove(power uint, category MoveCategory, moveType Type) MoveId {
 
 var TestMoveDefault = NewMove(10, MoveCategoryPhysical, TypeNormal)
 var TestMoveNoDamage = NewMove(0, MoveCategoryPhysical, TypeNormal)
+
+var _ = Describe("Move PP Consumption", func() {
+	a := Agent(new(dumbAgent))
+	It("should decrement move's PP by 1 when used", func() {
+		b := New1v1Battle(
+			GeneratePokemon(PkmnSquirtle, WithMoves(MoveSplash)), &a,
+			GeneratePokemon(PkmnSquirtle, WithMoves(MoveSplash)), &a,
+		)
+		b.rng = AlwaysRNG()
+		Expect(b.Start()).To(Succeed())
+		t, _ := b.SimulateRound()
+		Expect(t).To(HaveTransactionsInOrder(
+			PPTransaction{
+				Amount: -1,
+			},
+			PPTransaction{
+				Amount: -1,
+			},
```

Comment:
Don't we need a `*Move` to validate these?

Reply:
Not really, since this is only testing to make sure the PP goes down when moves are used. It doesn't matter what the moves are.